ToDoList/src/ToDoList.WebApi/Controllers/ToDoItemsController.cs

```diff
@@ -0,0 +1,123 @@
+namespace ToDoList.WebApi.Controllers;
+
+using Microsoft.AspNetCore.Mvc;
+using ToDoList.Domain.DTOs;
+using ToDoList.Domain.Models;
+
+[ApiController]
+[Route("api/[controller]")]
+public class ToDoItemsController : ControllerBase
+{
+    private static readonly List<ToDoItem> Items = [];
+
+    [HttpPost]
+    public IActionResult Create(ToDoItemCreateRequestDto request)
+    {
+        //map to Domain object as soon as possible
+        var item = request.ToDomain();
+
+        //try to create an item
+        try
+        {
+            item.ToDoItemId = Items.Count == 0 ? 1 : Items.Max(o => o.ToDoItemId) + 1;
+            Items.Add(item);
+        }
+        catch (Exception ex)
+        {
+            return Problem(ex.Message, null, StatusCodes.Status500InternalServerError); //500
+        }
+
+        //respond to client
+        return Created(); //201
+    }
+
+    [HttpGet]
+    public IActionResult Read(ToDoItemReadRequestDto request)
+    {
+        //map to Domain object as soon as possible
+        var items = request.FromDomain();
+
+        //try to read a list of items
+        try
+        {
+            return (items == null || items.Count == 0) ? NotFound() : Ok(items);
+        }
+        catch (Exception ex)
+        {
+            return Problem(ex.Message, null, StatusCodes.Status500InternalServerError); //500
+        }
+    }
+
+    [HttpGet("{toDoItemId:int}")]
+    public IActionResult ReadById(int toDoItemId)
+    {
+        ToDoItem? item;
+
+        //try to read the item
+        try
+        {
+            item = Items.Find(o => o.ToDoItemId == toDoItemId);
+            return (item == null) ? NotFound() : Ok(item);
+        }
+        catch (Exception ex)
+        {
+            return Problem(ex.Message, null, StatusCodes.Status500InternalServerError); //500
+        }
+    }
+
+    [HttpPut("{toDoItemId:int}")]
+    public IActionResult UpdateById(int toDoItemId, [FromBody] ToDoItemUpdateRequestDto request)
+    {
+        var requestItem = request.ToDomain();
+        ToDoItem? itemToUpdate;
+
+        //try to find and update the item
+        try
+        {
+            itemToUpdate = Items.Find(o => o.ToDoItemId == toDoItemId);
+
+            if (itemToUpdate == null)
+            {
+                return NotFound();
+            }
+
+            //Tyto radky ted asi nejsou optimalni, protoze kdzy pouziju record, musim vyplnit vsechny properties.
+            //TODO: Nastavit defaults? Nebo pouzit radeji class? Urcite nechci pri updatu vyplnovat vse.
```

Comment:
ja by som naopak pri update vyplňoval vždy všetko. Je to také trochu fašistické, ale jedná sa o pár záznamov, nie je ich milión, a krásne ti to potom formalizuje celú komunikáciu: vždy vieš, čo máš od posielaného DTO čakať.

Reply:
Diky za super komentare. :) Zacnu tady. Myslela jsem to tak, ze nechci, aby koncovy uzivatel vyplnoval vsechna pole pri updatu, napr. detailni popis itemu. Takze to bych mohla udelat tak, ze si pomoci GET requestu zavolam data pro zobrazeni itemu v UI a tato data potom slouzi jako zaklad pro PUT request? Ze zmenim ta pole, ktere zmeni uzivatel v UI a ta ostatni nezmenena poslu na zaklade predchoziho GET requestu? Tim padem nebudu nutit uzivatele prepisovat vse a zaroven muzu pocitat s tim, ze vsechna pole v DTO jsou vyplnena?